src/backend/src/utils/change-requests.utils.ts

```diff
@@ -378,7 +378,7 @@ export const applyProjectProposedChanges = async (
       projectWbsElmeId = proj.wbsElementId;
     }
 
-    const promises = projectProposedChanges.workPackageProposedChanges.map(async (proposedChange) => {
+    for (const proposedChange of projectProposedChanges.workPackageProposedChanges) {
```

Comment:
yes mapping doesnt wait for the previous await to be finished before firing the next one, you need to use a for of loop for that functionality. Otherwise you enter a race condition with work package number